src/api/argo-templates/base/sat-general-template-iuf.argo.yaml

```diff
@@ -0,0 +1,99 @@
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+apiVersion: argoproj.io/v1alpha1
+kind: WorkflowTemplate
+metadata:
+  name: sat-general
+spec:
+  entrypoint: main
+  templates:
+    ### Main Steps ###
+    - name: main
+      inputs:
+        parameters:
+          - name: auth_token
+          - name: global_params
+          - name: script_content
+      steps:
+        - - name: sat-wrapper
+            template: sat-wrapper
+            arguments:
+              parameters:
+                - name: auth_token
+                  value: "{{inputs.parameters.auth_token}}"
+                - name: script_content
+                  value: "{{inputs.parameters.script_content}}"
+    ### Templates ###
+    ## sat-wrapper ##
+    - name: sat-wrapper
+      nodeSelector:
+        kubernetes.io/hostname: ncn-m001
+      tolerations:
+        - key: node-role.kubernetes.io/master
+          effect: NoSchedule
+      metadata:
+        annotations:
+          sidecar.istio.io/inject: "false"
+      inputs:
+        parameters:
+          - name: ca_certs_bundle
+            value: /etc/ssl/certs/ca-certificates.crt
+          - name: auth_token
+          - name: script_content
+      script:
+        image: artifactory.algol60.net/sat-docker/stable/cray-sat:3.20.0
+        command: [sh]
+        source: |
+          #!/bin/sh
+          ts="$(mktemp)"
+          cat <<'EOF' > "${ts}"
+
+          set -e
+          sat init
+
+          # Write the auth token parameter to the file
+          JSON_FMT='{"access_token":"%s"}\n'
+          printf "$JSON_FMT" "{{inputs.parameters.auth_token}}" > /root/.config/sat/token.json
+          echo -n $(tr -d "\n\r" < /root/.config/sat/token.json) > /root/.config/sat/token.json
+
+          # Set up the SAT config to use the token file
+          sed -i -e 's|.*token_file.*|token_file = "/root/.config/sat/token.json"|' /root/.config/sat/sat.toml
+
+          {{inputs.parameters.script_content}}
+
+          EOF
+          chmod +x ${ts}
+          bash -e ${ts}
```

Comment:
A couple nit-picks:
* I think `set -e` above and `bash -e` here are redundant.
* I think it's also possible to use the heredoc as the stdin to bash, instead of writing a temp file, with something like the following, though I'm not sure on the exact syntax:
```bash
cat <<EOF | bash -
<script source>...
EOF
```

With such an approach, though, it might be good to encapsulate the wrapper code and the user-provided script in bash functions for safety, like the following:

```bash
cat <<EOF | bash -
function setup() {
  sat init
  JSON_FMT=...
  printf "$JSON_FMT" ...
  ... # Rest of setup wrapper code
}

function user_script() {
  {{inputs.parameters.script_content}}
}

function main() {
  setup
  user_script
}
main
```